### DISCUSSION TITLE
Is it possible to update or create multiple object of the same type at once?

### DISCUSSION THREAD

**Discussion by bouchtaoui-dev** [pocketbase/pocketbase#3333] [Q&A] [answered]
Hello there,

Again, what a great app and thank you for your effort and time!

Is it possible to update multiple object of the same type at once? For example, I have multiple todo tasks and sometimes I change the order of the tasklist based on priority. This means the indexes of the tasks change, so I need to update the tasks with the new indexes on PocketBase.

Now, I have to do something like this in a loop:

```js
const record = await pb.collection('tasks').update('RECORD_ID', data);
```
_I could do this concurrently, but still..._

Thank you!

**Comment by ganigeorgiev** [comment, marked as answer]
Unfortunately bulk/transactional operations are not supported via the web APIs at the moment (_this exists in the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1) as part of #48 but it is pushed aside for now_).

---

> Now, I have to do something like this in a loop:
> `const record = await pb.collection('tasks').update('RECORD_ID', data);`
> I could do this concurrently, but still...

_2 side-notes:_

- You probably are already aware of this but note that `await` will cause the requests to execute one by one if called in a regular loop. If you want the requests to be executed concurrently, you can use [`Promise.all`/`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all):
    ```js
    const promises = [];

    const items = [...]
    for (let item of items) {
        promises.push(pb.collection('tasks').update(item.id, item));
    }

    await Promise.all(promises)
    ```
- [Auto cancellation](https://github.com/pocketbase/js-sdk#auto-cancellation) will be triggered if executing concurrent _create_ requests to the same collection because all requests will share the same request key (by default the request path; this is not an issue with the _update_ because the path would be different for each record). To avoid eventual abort errors, you can disable the auto cancellation for the specific create requests by using the `requestKey:null` option:
    ```js
    const promises = [];

    for (let item of items) {
        promises.push(pb.collection('tasks').create(item, {
            requestKey: null, // disables auto cancellation for this particular request
        }));
    }

    await Promise.all(promises)
    ```
